dev-requirements.txt

```
@@ -5,4 +5,5 @@ mypy==0.901
 pylint==2.6.0
 pytest==6.1.2
 pytest-cov==2.11.1
-types-requests==2.25.0
\ No newline at end of file
+qiskit-terra~=0.18.3
+types-requests==2.25.0
```

should be
```suggestion
types-requests>=2.26.0
```
to match the `requests` version in applications-superstaq (this also needs to be corrected in [applications-superstaq's dev-requirements.txt](https://github.com/SupertechLabs/applications-superstaq/blob/main/dev-requirements.txt))

---

alternatively what do you think about changing `check/mypy` to include the flags `--install-types --non-interactive`? That way we could get rid of all `types-*` requirements in `*superstaq` and not worry about these cross-repo version consistencies

---

(also this comment isn't especially pertinent to this PR so feel free to ignore and I'll create a new issue)